Cargo.toml

```diff
@@ -28,6 +28,7 @@ toml = "~0.2.1"
 env_logger = "~0.3.4"
 typed-arena = "~1.2.0"
 clap = "~2.19.0"
+parking_lot = "~0.3.6"
```

Comment:
Not having to deal with possible poison errors here is kind of nice, and the parking_lot RwLock has far better perf than the std RwLock.

Seems like the current usage doesn't have any potential poisoning issues though. Maybe the unwraps would be ok.